library/core/src/main/java/com/google/android/exoplayer2/source/ConcatenatingMediaSource.java

```diff
@@ -297,6 +304,79 @@ public final synchronized void removeMediaSource(
     }
   }
 
+  /**
+   * Removes a range of {@link MediaSource}s from the playlist, by specifying an initial index
+   * (included) and a final index (excluded).
+   *
+   * <p>Note: when specified range is empty, no actual media source is removed and no exception
+   * is thrown.
+   *
+   * @param fromIndex The initial range index, pointing to the first media source that will be
+   *     removed. This index must be in the range of 0 &lt;= index &lt; {@link #getSize()}.
+   * @param toIndex The final range index, pointing to the first media source that will be left
+   *     untouched. The last media source to be removed is at index {@code toIndex} - 1.
+   *     This index must be in the range of 0 &lt;= index &lt; {@link #getSize()}.
+   *
+   * @throws IndexOutOfBoundsException when either index is out of playlist bounds, i.e. {@code
+   *      fromIndex} &lt; 0 or &gt;= {@link #getSize()}, {@code toIndex} &lt; 0 or &gt; {@link
+   *      #getSize()}
+   * @throws IndexOutOfBoundsException when range is malformed, i.e. {@code fromIndex} &gt;
+   *      {@code toIndex}
+   */
+  public final synchronized void removeMediaSourceRange(int fromIndex, int toIndex) {
+    removeMediaSourceRange(fromIndex, toIndex, null);
+  }
+
+  /**
+   * Removes a range of {@link MediaSource}s from the playlist, by specifying an initial index
+   * (included) and a final index (excluded), and executes a custom action on completion..
+   *
+   * <p>Note: when specified range is empty, no actual media source is removed and no exception
+   * is thrown.
+   *
+   * @param fromIndex The initial range index, pointing to the first media source that will be
+   *     removed. This index must be in the range of 0 &lt;= index &lt; {@link #getSize()}.
+   * @param toIndex The final range index, pointing to the first media source that will be left
+   *     untouched. The last media source to be removed is at index {@code toIndex} - 1.
+   *     This index must be in the range of 0 &lt;= index &lt; {@link #getSize()}.
+   *
+   * @throws IndexOutOfBoundsException when either index is out of playlist bounds, i.e. {@code
+   *      fromIndex} &lt; 0 or &gt;= {@link #getSize()}, {@code toIndex} &lt; 0 or &gt; {@link
+   *      #getSize()}
+   * @throws IndexOutOfBoundsException when range is malformed, i.e. {@code fromIndex} &gt;
+   *      {@code toIndex}
+   * @param actionOnCompletion A {@link Runnable} which is executed immediately after the media
+   *     source range has been removed from the playlist.
+   */
+  public final synchronized void removeMediaSourceRange(
+          int fromIndex, int toIndex, @Nullable Runnable actionOnCompletion) {
+    if (fromIndex < 0 || fromIndex >= mediaSourcesPublic.size()) {
```

Comment:
No need to do the explicit bound checks here. As soon as you call the subList method, the IndexOutOfBoundsException will be thrown anyway.